tuning/utils/preprocessing_utils.py

```diff
@@ -29,8 +29,55 @@
 logger = logging.get_logger("sft_trainer_preprocessing")
 
 
+# check if the provided dataset is pretokenized or not
+# the check is taken from trl
+# https://github.com/huggingface/trl/blob/ddf4c8dc3ecf6d9ee2b24f94c62182ffd682c808/trl/trainer/sft_trainer.py#L498-L509
+def is_pretokenized_dataset(data_path: str):
+    if data_path:
+        # load one sample from the dataset in order to inspect columns
+        dataset = datasets.load_dataset("json", data_files=data_path, split="train[:1]")
```

Comment:
Can you add graceful handling for if the dataset is empty?

Reply:
nesting it from the same error class `DatasetGenerationError` since there are tests on malformed data already that assert on DatasetGenerationError Thanks. let me know if the message has to be refined. Thanks